### ISSUE TITLE
The DKMS kernel hook fails completely if a single module fails

### ISSUE THREAD

**Issue by ericonr** [void-linux/void-packages#23124]
### Expected behavior

If a DKMS module fails to build, all other modules should still be built.

### Actual behavior

If a DKMS module (such as NVIDIA for `linux5.7`) fails to build, it aborts the whole operation, skipping the attempt to build other modules, such as ZFS.

### Steps to reproduce the behavior

Have `linux5.7`, `linux5.7-headers`, `nvidia-dkms` and `zfs` installed. Try to run

```
# xbps-reconfigure -f linux5.7
linux5.7: configuring ...
Executing post-install kernel hook: 10-dkms ...
Available DKMS module: nvidia-440.82.
Available DKMS module: zfs-0.8.4.
Building DKMS module: nvidia-440.82... FAILED!
Executing post-install kernel hook: 20-dracut ...
Executing post-install kernel hook: 20-dracut-uefi ...
Executing post-install kernel hook: 40-sbsigntool ...
Executing post-install kernel hook: 50-efibootmgr ...
Executing post-install kernel hook: 50-gummiboot ...
Executing post-install kernel hook: 50-refind ...
linux5.7: configured successfully.
```

**Comment by ericonr** [comment]
So, this seems to be due to how the 10-dkms hook is written:

```
	if [ $(echo "$status"|grep -c ": built") -eq 0 ]; then
		# Check if the module is still there.
		if [ ! -f usr/src/${module}-${modulever}/dkms.conf ]; then
			echo "Skipping unexistent DKMS module: ${module}-${modulever}."
			shift 2
			continue
		fi
		# Build the module
		echo -n "Building DKMS module: ${module}-${modulever}... "
		dkms build -q -m ${module} -v ${modulever} -k ${VERSION} -a ${ARCH}
		rval=$?
		if [ $rval -eq 9 ]; then
			echo "skipped!"
			shift; shift
			continue
		elif [ $rval -eq 0 ]; then
			echo "done."
		else
			echo "FAILED!"
			exit $?
		fi
		status=$(dkms status -m ${module} -v ${modulever} -k ${VERSION})
	fi
```

If building a single module fails, it will exit instead of trying the others. On the other hand, the `dkms` trigger for XBPS has

```
	for f in $(echo lib/modules/*); do
		_kver=$(basename $f)
		if [ ! -e "${f}/build/include" ]; then
			echo "Skipping kernel-${_kver}. kernel-headers package not installed..."
			continue
		fi
		if [ ! -f ${f}/build/scripts/basic/fixdep ] ; then
			echo -n "Building scripts for kernel-${_kver}... "
			yes "" | make -j$(nproc) -C ${f}/build scripts > ${f}/build/make.log 2>&1
			if [ $? -eq 0 ]; then
				echo "done."
			else
				echo "FAILED!"
				echo "Kernel scripts failed to build, please check /${f}/build/make.log"
				echo "for errors in the log file."
			fi
		fi
		set -- ${dkms_modules}
		while [ $# -gt 0 ]; do
			echo -n "Building DKMS module '$1-$2' for kernel-${_kver}... "
			$DKMS build -m "$1" -v "$2" -k "${_kver}" >/dev/null 2>&1
			if [ $? -eq 0 ]; then
				echo "done."
			else
				echo "FAILED!"
				echo "DKMS module '$1-$2' failed to build, please check /var/lib/dkms"
				echo "for errors in the log file."
				shift 2; continue
			fi
			echo -n "Installing DKMS module '$1-$2' for kernel-${_kver}... "
			$DKMS install -m "$1" -v "$2" -k "${_kver}" >/dev/null 2>&1
			if [ $? -eq 0 ]; then
				echo "done."
			else
				echo "FAILED!"
				echo "DKMS module '$1-$2' failed to install, please do this manually"
				echo "and check for errors in the log file."
			fi
			shift 2
		done
```

which, after a failure, just continues trying to build others. This behavior for the kernel hook seems to be from 2011, so it's just something that went under the radar.

I think it makes sense to try to use the same logic in both, then.

**Comment by ahesford** [comment]
Fixed with commit 2bec57b6.
